### ISSUE TITLE
Add WikiText syntax for var, samp and kbd elements

### ISSUE THREAD

**Issue by nameanyone** [Jermolene/TiddlyWiki5#2231]
Feature request.

**Comment by pmario** [comment]
what would you suggest?

**Comment by nameanyone** [comment]
It's not easy, most characters are already being used. For starters, how about 

`%E%=%m%%c%^^2^^`

`~~samp inline~~`

```
~ ~~
samp block
(without a space, I just had to add it here)
~ ~~
```
```
Hit ""Ctrl""+""Alt""+""Del""
```

**Comment by nameanyone** [comment]
@pmario, I updated my first comment with suggestions.

**Comment by felixhayashi** [comment]
Hi @nameanyone I think wikitext makes most sense for elements frequenly used in everyday "wiki work" and to simplify the syntax, plus making it look prettier.

I don't think all html elements should be translated into wikitext. In my opinion `<kbd>`, from an end user perspective, is quite simple, not much used in daily work and you don't need to invent a way to write this in wikitext. Also every new wikitext rule introduced makes it more complicated to invent new rules in the future as less "potential syntax" is available. So I think we should only have wikitext for important, frequently used stuff. But maybe others think differently about it.

**Comment by nameanyone** [comment]
My "wiki work" is primarily documenting computer stuff, so I would use these regularly if they were available.

**Comment by felixhayashi** [comment]
@nameanyone while this is not issue related, maybe danielo's keyboard shortcuts plugin can support you a bit here (https://github.com/danielo515/TW5-KeyboardSnippets). Allows you to create and customize keykombos.

**Comment by tobibeer** [comment]
What are **var**, **samp** or **kbd**?

**Comment by felixhayashi** [comment]
I only know `<kbd>`. It's an html tag. https://developer.mozilla.org/en/docs/Web/HTML/Element/kbd

**Comment by tobibeer** [comment]
I'd rather see `%foo%` be reserved for textual placeholders.

**Comment by tobibeer** [comment]
> Hi @nameanyone I think wikitext makes most sense for elements frequenly used in everyday "wiki work" and to simplify the syntax, plus making it look prettier.
>
> I don't think all html elements should be translated into wikitext. 

:+1:

**Comment by tobibeer** [comment]
I'd think your best approach would be to use macros, e.g.:

* `<<out "foo">>`
* `<<kbd Z>>`
* `<<var E>>`

This for example cannot work:

`$samp inline$`

Because the **$foo$** syntax is already used for macro parameter substitution.

**Comment by nameanyone** [comment]
> $foo$ syntax is already used for macro parameter substitution

Yes, how did I miss that! I removed it from the top comment. The characters can be anything, let's not pick on them. The idea here is to generate [semantic HTML](https://en.wikipedia.org/wiki/Semantic_HTML)

**Comment by pmario** [comment]
@rubaboo 

I think the way to go, is with macros, as tobias suggested. It's the easiest way to achieve, what you want out of the box in a compatible way.
